### PR TITLE
doc: add architecture-design doc

### PR DESCRIPTION
**Pull request by neverchanje**
## Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests

### CODE REVIEW

**Review by fuyufjh — APPROVED** [review]

File: docs/architecture-design.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+This doc serves as one of the materials for newcomers to learn the high-level architecture and the functionalities of each component.
+
+## Architecture 
+
+There are currently 3 types of nodes in the cluster: 
+
+* **Frontend**: Frontend is a stateless proxy that accepts user queries through Postgres protocol. It is responsible for parsing, validation, optimization, and answering the results of each individual query. 
+* **ComputeNode**: ComputeNode is responsible to execute the optimized query plan. 
+* **MetaServer**: The central metadata management service. It also acts as a failure detector that periodically sends a heartbeat to every Frontends and ComputeNode-s in the cluster.
```
[the remark this answers is not in the document]
Comment (fuyufjh) [reply docs/architecture-design.md]: So shall we replace "ComputeNode-s" to "compute nodes"?

[the remark this answers is not in the document]
Comment (hengm3467) [reply docs/architecture-design.md]: I think we can say "frontend nodes", "compute nodes", and "meta nodes" (do we have more than one meta nodes in one cluster?)

I think we can move the content in this document to the "Architecture" article (https://www.risingwave.dev/docs/latest/architecture/) in our user docs. Currently, the Architecture page is a bit too simple.

[the remark this answers is not in the document]
Comment (fuyufjh) [reply docs/architecture-design.md]: cc @neverchanje

**Comment by jon-chuang** [comment]
@neverchanje the svg https://github.com/singularity-data/risingwave/blob/06bcfa918249e8c535bae7a2f9c98a28465be156/docs/images/architecture-design/batch-query.svg has the name bummock instead of hummock

**Pull request merged**